src/Workspaces/CSharp/Portable/CodeGeneration/NamedTypeGenerator.cs

```diff
@@ -114,29 +117,22 @@ internal static class NamedTypeGenerator
                 // remove the primary constructor from the list of members to generate.
                 members = members.Remove(primaryConstructor);
 
-                // remove any properties that were created by the primary constructor
-                members = members.WhereAsArray(m => m is not IPropertySymbol property || !primaryConstructor.Parameters.Any(p => p.Name == property.Name));
+                // remove any fields/properties that were created by the primary constructor
+                members = members.WhereAsArray(m => m is not IPropertySymbol and not IFieldSymbol || !primaryConstructor.Parameters.Any(p => p.Name == m.Name));
```

Comment:
nit: Checking symbol kind may be cheaper.

Reply:
In general, this is during the fix part, and will be run on tuples containing a reasonable amount of elements. So this will be fine from a perf perspective.